### ISSUE TITLE
Nano: Tensorflow trace(onnx/openvino) api discussion

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#6006]
Well the question here is easy.

Should we put it as before in `model.quantize` and `model.trace`? or we should move all the previous and newly added features to `InferenceOptimizer`'s `trace` and `quantize`.

I would prefer the second choice so that in future we could
1. Use `patch_nano` to handle (nearly) all the training speed-up.
2. Use `InferenceOptimizer` to handle all the inference speed-up.

**Comment by TheaperDeng** [comment]
@rnwang04 @MeouSker77 any thoughts?

**Comment by TheaperDeng** [comment]
> One question here, if we put all the previous and newly added tf features to `InferenceOptimizer`'s `trace` and `quantize`, it seems that this will lead to the tight coupling between Pytorch and Tensorflow ? Is this against our existing installation method: `pip install bigdl nano [tensorflow]` and `pip install bigdl nano [pythoch]`?

Yeah, good question. We may use something like: https://github.com/intel-analytics/BigDL/blob/main/python/chronos/src/bigdl/chronos/utils.py#L39-L93 here to deal with the tf/torch dependency control

**Comment by MeouSker77** [comment]
> Yeah, good question. We may use something like: https://github.com/intel-analytics/BigDL/blob/main/python/chronos/src/bigdl/chronos/utils.py#L39-L93 here to deal with the tf/torch dependency control

It's a possible solution, or maybe we can just create a new `InferenceOptimizer` for tensorflow?

**Comment by jason-dai** [comment]
Maybe we should add an `InferenceOptimizer` for Keras too.
